keyring/backends/SecretService.py

```diff
@@ -31,6 +31,11 @@ class Keyring(KeyringBackend):
     @properties.ClassProperty
     @classmethod
     def priority(cls):
+        try:
+            from jeepney.bus_messages import message_bus
+            from jeepney.integrate.blocking import Proxy
+        except ImportError:
+            raise RuntimeError("jeepney required")
```

Comment:
Could this functionality be exposed by secretstorage? I don't understand the relationship between secretstorage and jeepney, but keyring seems to have a dependency on both jeepney and secretstorage, even though this change adds the only reference to jeepney.

Reply:
> I don't understand the relationship between secretstorage and jeepney

Jeepney handles D-Bus communications, secretstorage uses that to talk to the secrets service. As an analogy, a Github API client could be built on top of requests, which handles HTTP communications.

It would certainly be possible for SecretService to have a `check_available()` function or similar that encapsulated this logic. Up to @mitya57 whether he wants that. :-)

Reply:
I added a `check_service_availability()` function to SecretStorage now. I bumped the dependency to 3.2, but the new version is not released yet (I will do it after review).

A question to @takluyver: Do you know if this code can raise any exceptions, provided that the connection was established successfully? Currently keyring only captures `SecretStorageException`, but in that new function I do not convert any exceptions from Jeepney to `SecretStorageException`. I can use my wrapper class `DBusAddressWrapper` to do that, though.

https://github.com/mitya57/secretstorage/blob/master/secretstorage/__init__.py#L84-L90

Reply:
> Do you know if this code can raise any exceptions, provided that the connection was established successfully? 

In theory, it could: the message bus could respond with an error, which Jeepney would turn into a `DBusErrorResponse` exception. Or if it sends back data that's not a valid D-Bus message, or there's a bug in Jeepney's parsing code, there could be all sorts of exceptions. In practice, I think it's unlikely - the message bus is generally reliable (it has to be), and we're not trying to do anything complex with it.

Reply:
Ok, I left the code as is then. If we get any bugs about such exceptions, I will catch them and make that function return False in those cases.